config/config.go

```diff
@@ -154,25 +154,29 @@ type VPCProviderConfig struct {
 	VPCBlockProviderName string `toml:"vpc_block_provider_name" envconfig:"VPC_BLOCK_PROVIDER_NAME"`
 	VPCBlockProviderType string `toml:"provider_type"`
 
-	EndpointURL      string `toml:"gc_riaas_endpoint_url"`
-	TokenExchangeURL string `toml:"gc_token_exchange_endpoint_url"`
-	APIKey           string `toml:"gc_api_key" json:"-"`
-	ResourceGroupID  string `toml:"gc_resource_group_id"`
-	VPCAPIGeneration int    `toml:"vpc_api_generation" envconfig:"VPC_API_GENERATION"`
-	APIVersion       string `toml:"api_version,omitempty" envconfig:"VPC_API_VERSION"`
+	EndpointURL        string `toml:"gc_riaas_endpoint_url"`
+	PrivateEndpointURL string `toml:"gc_riaas_endpoint_private_url"`
+	TokenExchangeURL   string `toml:"gc_token_exchange_endpoint_url"`
+	APIKey             string `toml:"gc_api_key" json:"-"`
+	ResourceGroupID    string `toml:"gc_resource_group_id"`
+	VPCAPIGeneration   int    `toml:"vpc_api_generation" envconfig:"VPC_API_GENERATION"`
+	APIVersion         string `toml:"api_version,omitempty" envconfig:"VPC_API_VERSION"`
 
 	//NG Properties
-	G2EndpointURL      string `toml:"g2_riaas_endpoint_url"`
-	G2TokenExchangeURL string `toml:"g2_token_exchange_endpoint_url"`
-	G2APIKey           string `toml:"g2_api_key" json:"-"`
-	G2ResourceGroupID  string `toml:"g2_resource_group_id"`
-	G2VPCAPIGeneration int    `toml:"g2_vpc_api_generation" envconfig:"G2_VPC_API_GENERATION"`
-	G2APIVersion       string `toml:"g2_api_version,omitempty" envconfig:"G2_VPC_API_VERSION"`
+	G2EndpointURL        string `toml:"g2_riaas_endpoint_url"`
+	G2EndpointPrivateURL string `toml:"g2_riaas_endpoint_private_url"`
+	G2TokenExchangeURL   string `toml:"g2_token_exchange_endpoint_url"`
+	G2APIKey             string `toml:"g2_api_key" json:"-"`
+	G2ResourceGroupID    string `toml:"g2_resource_group_id"`
+	G2VPCAPIGeneration   int    `toml:"g2_vpc_api_generation" envconfig:"G2_VPC_API_GENERATION"`
+	G2APIVersion         string `toml:"g2_api_version,omitempty" envconfig:"G2_VPC_API_VERSION"`
 
 	Encryption      bool   `toml:"encryption"`
 	VPCTimeout      string `toml:"vpc_api_timeout,omitempty" envconfig:"VPC_API_TIMEOUT"`
 	MaxRetryAttempt int    `toml:"max_retry_attempt,omitempty" envconfig:"VPC_RETRY_ATTEMPT"`
 	MaxRetryGap     int    `toml:"max_retry_gap,omitempty" envconfig:"VPC_RETRY_INTERVAL"`
+	// IKSTokenExchangePrivateURL, for private cluster support hence using for all cluster types
+	IKSTokenExchangePrivateURL string `toml:"iks_token_exchange_endpoint_private_url"`
```

Comment:
Can we make this as `IKSTokenExchangePrivateURL string `toml:"iks_token_exchange_endpoint_private_url" toml:"gc_token_exchange_endpoint_private_url" toml:"g2_token_exchange_endpoint_private_url"`

For third party vendor , they can put the private endpoint in one of these three properties

Reply:
I did not see this is working